pkg/booter/localbooter.go

```diff
@@ -0,0 +1,102 @@
+package booter
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+// LocalBooter implements the Booter interface for booting from local storage.
+type LocalBooter struct {
+	Type       string `json:"type"`
+	Method     string `json:"method"`
+	DeviceGUID string `json:"device_guid"`
+	Kernel     string `json:"kernel,omitempty"`
+	KernelArgs string `json:"kernel_args,omitempty"`
+	Initramfs  string `json:"ramfs,omitempty"`
+}
+
+// NewLocalBooter parses a boot entry config and returns a Booter instance, or
+// an error if any
+func NewLocalBooter(config []byte) (Booter, error) {
+	/*
+		The configuration format for a LocalBooter entry is a JSON with the following structure:
+
+		{
+			"type": "localboot",
+			"method": "<method>",
+			"device_guid": "<device GUID or empty>"
+			"kernel": "<kernel path or empty>",
+			"kernel_args": "<kernel args or empty>",
+			"ramfs": "<ramfs path or empty>",
+		}
+
+		`type` is always set to "localboot"
+		`method` can be either "grub" or "path".
+		    The "grub" method will look for grub.cfg or grub2.cfg on the specified device.
+		    If no device is specified, it will look on all the attached storage devices,
```

Comment:
This sounds like you can use grub mode and still specify which device the grub is installed on. (but the code above always scans all devices instead) This feature would actually be very handy! For some big filesystems, it can even be very expensive to mount them all (double-digit _minutes_). In many cases, people using grub will know where its configuration is, so I think we should support this.

Reply:
That's an inconsistency, I will fix it, thanks for spotting!

Reply:
and will add a `guid` parameter to the GRUB mode too